technical-docs/designs/infra/managed-identity.md

```diff
@@ -0,0 +1,251 @@
+`azure-common-infra` and will contain the following:
+
+- Managed Identity System Level for AKS
+- Pod Identity Security Policy
+- Backend State
+
+**Sample `Main.tf`**
+
+```
+resource "azurerm_resource_group" "aks_rg" {
+  name     = local.aks_rg_name
+  location = local.region
+}
+
+module "aks-gitops" {
+  source = "github.com/microsoft/bedrock?ref=aks_msi_integration//cluster/azure/aks-gitops"
+
+  acr_enabled              = true
+  agent_vm_count           = var.aks_agent_vm_count
+  agent_vm_size            = var.aks_agent_vm_size
+  cluster_name             = local.aks_cluster_name
+  dns_prefix               = local.aks_dns_prefix
+  flux_recreate            = var.flux_recreate
+  gc_enabled               = true
+  msi_enabled              = true
+  gitops_ssh_url           = var.gitops_ssh_url
+  gitops_ssh_key           = var.gitops_ssh_key_file
+  gitops_path              = var.gitops_path
+  gitops_poll_interval     = var.gitops_poll_interval
+  gitops_label             = var.gitops_label
+  gitops_url_branch        = var.gitops_url_branch
+  kubernetes_version       = var.kubernetes_version
+  resource_group_name      = azurerm_resource_group.aks_rg.name
+  service_principal_id     = module.app_management_service_principal.service_principal_application_id
+  service_principal_secret = module.app_management_service_principal.service_principal_password
+  ssh_public_key           = file(var.ssh_public_key_file)
+  vnet_subnet_id           = module.vnet.vnet_subnet_ids[0]
+  network_plugin           = var.network_plugin
+  network_policy           = var.network_policy
+  oms_agent_enabled        = var.oms_agent_enabled
+}
+```
+
+Questions & Limitations:
+
+- With the deployment of the `azure-common-infra` template for Key Vault, will
+  that also need to be modified for Manage Identity to whitelist AKS to access
+  keyvault?
+
+### 3.3 Testing for Managed Identity enable AKS Gitops Environments
+
+The testing for the Managed Identity enabled AKS gitops environment will
+incorporate the aforementioned new Infrastructure DevOps Model Flow for
+Terraform to assess pod identity access for a Voting App service deployed using
+terraform and a flux manifest repository.
+
+#### Unit Tests
+
+Cobalt Test Fixtures includes a library that simplifies writing unit terraform
+tests against templates. It extracts out pieces of this process and provides a
+static validation for a json sample output per module. For this, we require Unit
+Tests for the following modules:
+
+- AKS
+- Key Vault
+- VNet
+- Subnet
+- Gitops
+
+#### Integration Tests
+
+Integration tests will validate resource interoperability upon deployment.
+Pending a successful `terraform apply`, using a go script and terratest go
+library, this design will create an integration test for the respective
+environment template that verifies
+
+- Flux namespace
+- Access to cluster using Pod Identity
+- Access to key using flex-volume
+- 400 response on Voting App
+
```

Comment:
shouldnt we also test for a 200 response?

Reply:
presumably the voting app K8 deployment resource will be binded to a pod identity, correct?

Reply:
Changed 400 --> 200 and modified success parameters for integration test